src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -106,7 +106,21 @@ private function parseImports(\DOMDocument $xml, $file)
         $defaultDirectory = dirname($file);
         foreach ($imports as $import) {
             $this->setCurrentDir($defaultDirectory);
-            $this->import($import->getAttribute('resource'), XmlUtils::phpize($import->getAttribute('type')) ?: null, (bool) XmlUtils::phpize($import->getAttribute('ignore-errors')), $file);
+
+            $resourceFile = $import->getAttribute('resource');
```

Comment:
Two notes here:
- tests still pass if I revert this change on XmlFileLoader on my setup
- I think this logic should be handled at the FileLoader level: I don't see why both Xml&Yaml loaders should care about that. They should be untouched to me.